### PR TITLE
(MODULES-3588) Update documentation for change in MODULES-3399

### PR DESCRIPTION
**Pull request by glennsarti**
The behavior about $LASTEXITCODE was modified in MODULES-3399 however there
was no documentation to demonstrate the change.  This PR modifies the README
with the new behavior and example code.

[ci skip]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,6 +77,37 @@ $obj.Rename("OtherGuest")
...
+and we use the following Puppet manifest
+
+~~~ puppet
+exec { 'test':
+  command   => '& C:\fail.ps1',
+  provider  => powershell,
+}
+~~~
+
+The `exec['test']` resource will always fail because the exit code from the the external file `C:\fail.ps1` is `1`.  This behavior may have unintended consequences if you are combining mulitple external files.
```
Comment (Iristyle) [inline README.md]: `multiple`

File: README.md [hunk 2]
```diff
@@ -76,6 +77,37 @@ $obj.Rename("OtherGuest")
 
 This has the added benefit of not requiring escaping '$' in the PowerShell code. Note that the files need to have DOS linefeeds or they will not work as expected. One tool for converting UNIX linefeeds to DOS linefeeds is [unix2dos](http://freecode.com/projects/dos2unix).
 
+### External files and exit codes
+If you are calling external files, such as other PowerShell scripts or executables, be aware that the last executed script's exitcode will be used by Puppet to determine whether the command was succesful.  For example:
```
Comment (Iristyle) [inline README.md]: `successful`

File: README.md [hunk 3]
```diff
@@ -76,6 +77,37 @@ $obj.Rename("OtherGuest")
...
+
+Suppose the file `C:\fail.ps1` contains the following PowerShell script
+
+~~~ powershell
```
Comment (Iristyle) [inline README.md]: Without getting too carried away, I think we should have a multi-line script sample that calls an external executable multiple times that clearly demonstrates the last command being executed as the overriding exit code value.

Comment (glennsarti) [inline README.md]: Updated the script to call CMD with different exit codes

**Comment by Iristyle** [comment]
I think we might want to clarify a bit more / add further detail in a subsequent iteration, but this is probably fine for now.  (For instance, it might help to talk about the difference between cmdlets and external executables / the call operator - and how cmdlets don't have a `$LASTEXITCODE` but that external execs will)

I'm going to merge and docs review can take another pass to make sure its clear enough.

**Comment by Iristyle** [comment]
Changed branch to stable, so that this can be merged up.

**Pull request merged**